app/assets/javascripts/discourse/app/routes/topic.js

```diff
@@ -199,9 +200,11 @@ const TopicRoute = DiscourseRoute.extend({
 
   @action
   showGrantBadgeModal() {
-    showModal("grant-badge", {
-      model: this.modelFor("topic"),
```

Comment:
I didn't see anywhere in the modal component nor the mixin where this is a dependency, it may have been something missed out from the original PR: https://github.com/discourse/discourse/pull/5498.